src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -124,7 +124,31 @@ static int32_t peep_move_one_tile(Direction direction, Peep* peep)
     peep->DestinationTolerance = 2;
     if (peep->State != PEEP_STATE_QUEUING)
     {
-        peep->DestinationTolerance = (scenario_rand() & 7) + 2;
```

Comment:
Why get rid of this bit as well? I think keeping the tolerance as well would be best. Then you can clearly mark where your changes diverge from vanilla.

Reply:
We *could* do, but I think that getting rid of it makes it easier to reason about the movement, as well as being better for performance. Instead of having two random factors to take into account for each tile move, you only have one. 

I'm also not sure that the resulting range of values is acceptable; if you roll a 9 here, and a -3 to offset below, the peep will reach their target a full 12 units away from the centre of the tile. I think this will lead to peeps being too far away from the centre line and walking off the sides of paths.

In principle I suppose we could keep this and drop the offset variable below, but I'm not a fan - DestinationTolerance is inherently 2D, but the point of this change is that we only want to allow a random tolerance in one dimension.

Reply:
I think having some variability in the destination tolerance makes for a tiny bit of more natural movement.

Reply:
Can you please clarify EXACTLY what you are expecting by "more natural" movement so I can go and test to see if you are right?

I am still convinced that varying the DestinationTolerance _in addition to_ the target position is going to do nothing but hurt performance.

Reply:
Calling scenario_rand is not an expensive operation so I don't see there being any issue with calling that more times and even if there was then you could get the random value once and use it for both the x/y offset and the tolerance. Have you tested it with the original line reinstated as well as your change to see how it would look?

Reply:
This is what it looks like with the original line reinstated:

![ezgif com-optimize](https://user-images.githubusercontent.com/588194/94371658-20dfe200-00c6-11eb-8336-cf435df1a579.gif)

As you can see, this results in peeps now being too spread away from the centre lines of the paths, to the point that they walk on the wrong sides of bridge railings etc.